gui/default/syncthing/device/editDeviceModalView.html

```diff
@@ -60,9 +60,12 @@ <h4 class="modal-title">
                 <div class="row">
                   <div class="col-md-4" ng-repeat="folder in folderList()">
                     <div class="checkbox">
-                      <label>
+                      <label ng-if="folder.label.length == 0">
```

Comment:
This could be an ng-switch or an if-else or something I think. Maybe it's not cleaner that way, but please check if it is and report back either way. :)